fl4health/preprocessing/autoencoders/vae_training.py

```diff
@@ -0,0 +1,51 @@
+from typing import Tuple
```

Comment:
Maybe you guys already discussed this a bit and decided against it. However, it feels like this transformer is a bit restricted by the transform functionality of the Dataset class. I wonder if its better for us to create an AutoEncoder dataset class where we have `self.data` and `self.target` attributes and we check that they are of equal dimension during the construction. This class could have an optional `self.conditional_data: Optional[torch.Tensor]` input and it if isn't none, then it could be concatenated with self.data. Perhaps this is a bad idea, but it would simplify a few things that I can see.

Reply:
We can create a converter if we want to, that takes a regular dataset and transforms it into an AutoEncoder dataset, sort of similar to this one, if we want as well.